File: JavaScript/Structural/Adapter.js
/* Adapter is a structural design pattern that allows 
objects with incompatible interfaces to collaborate. */

// Creating two incompatible classes
class FirstClass {
    firstProperty = 0;
    secondProperty = 1;
}

class SecondClass {
    firstProperty = '';
    secondProperty = '';
}

// creating the adapter to pass first class to second
class Adapter {
    constructor() {
        if (this.constructor === AbstractFactory) {
            throw new TypeError("Can't instantiate abstract class!!!");
        }
    }

    static adaptFirstToSecond(firstObj) {
        let secondObj = new SecondClass();
        secondObj.firstProperty = `${firstObj.firstProperty}`;
        secondObj.secondProperty = `${firstObj.secondProperty}`;
        return secondObj;
    }
}

// creating an object and adapting it
let firstObj = new FirstClass();
let secondObj = Adapter.adaptFirstToSecond(firstObj);
console.log(secondObj.secondProperty);

File: CSharp/Structural/Adapter.cs
namespace Structural;
/* Adapter is a structural design pattern that allows
objects with incompatible interfaces to collaborate. */

// Creating two incompatible classes
public class Adapter
{
    public class First
    {
        public int FirstProperty = 0;
        public int SecondProperty = 1;
    }

    public class Second
    {
        public string FirstProperty = "0";
        public string SecondProperty = "1";
    }

    public static class AdapterClass
    {
        public static Second AdaptFirstToSecond(First first)
        {
            var second = new Second();
            second.FirstProperty = first.FirstProperty.ToString();
            second.SecondProperty = first.SecondProperty.ToString();
            return second;
        }
    }
}